### ISSUE TITLE
LDAP login failing on 3.10.X

### ISSUE THREAD

**Issue by areis422** [eventum/eventum#1174]
**Version(s) affected**: 3.10.x (specifically 3.10.5 and 3.10.6)
**PHP Version**: 7.4.22 (didn't change between 3.9.2 and 3.10.6)

**Description**  
With LDAP enabled and working in 3.9.2, upgrading to 3.10.X results in the following message after submitting the login:
`
Sorry, there was an error logging you in: Call to a member function get() on null
`

**How to reproduce**  
1. have instance running on 3.9.2 with LDAP via Active Directory enabled.
2. Upgrade to 3.10.6
3. Fail on login

**Additional context**  
Log messages:
```
[2021-08-05 01:16:33] auth.INFO: Login attempt by 'user' was successful. {"user":"user","type":"success","extra":null} {"file":"/var/www/eventum-3.10.6/lib/eventum/class.auth.php","line":85,"class":"Auth","function":"saveLoginAttempt","memory_peak_usage":"4 MB","memory_usage":"4 MB","url":"/login.php/login.php","ip":"x.x.x.x","http_method":"POST","server":"staging.eventum.domain.tld","referrer":"https://staging.eventum.domain.tld/login.php"}
[2021-08-05 01:16:33] app.CRITICAL: Unable to instantiate auth adapter {"exception":"[object] (Error(code: 0): Call to a member function get() on null at /var/www/eventum-3.10.6/src/ServiceContainer.php:89)"} {"file":"/var/www/eventum-3.10.6/lib/eventum/class.auth.php","line":476,"class":"Auth","function":"getAuthBackend","memory_peak_usage":"2 MB","memory_usage":"2 MB","url":"/list.php","ip":"x.x.x.x","http_method":"GET","server":"staging.eventum.domain.tld","referrer":"https://staging.eventum.domain.tld/login.php"}
```

Auth config (from setup.php):
```php
    'auth' => [
        'adapter' => 'Eventum\\Auth\\Adapter\\ChainAdapter',
        'options' => [
            'Eventum\\Auth\\Adapter\\MysqlAdapter' => null,
            'Eventum\\Auth\\Adapter\\ChainAdapter' => [
                [
                    'Eventum\\Auth\\Adapter\\MysqlAdapter',
                    'Eventum\\Auth\\Adapter\\LdapAdapter',
                ],
            ],
            'Eventum\\Auth\\Adapter\\LdapAdapter' => null,
        ],
        'login_backoff' => [
            'count' => null,
            'minutes' => 15,
        ],
    ],
```

LDAP Configuration (redacted):
```php
    'ldap' => [
        'active_dn' => 'dc=domain,dc=tld',
        'inactive_dn' => 'ou=Disabled Users,OU=Users,OU=office,dc=domain,dc=tld',
        'default_role' => [
            1 => '6',
        ],
        'binddn' => 'user@domain.tld',
        'bindpw' => 'somepassword',
        'host' => 'domaincontroller.domain.tld',
        'port' => '636',
        'encryption' => 'ssl',
        'basedn' => 'dc=domain,dc=tld',
        'user_id_attribute' => 'sAMAccountName',
        'userdn' => 'DOMAIN\\%UID%',
        'user_filter' => '(&(sAMAccountName={username})(memberOf=CN=somegroup,OU=Roles,OU=Groups,OU=office,DC=domain,DC=tld))',
        'customer_id_attribute' => '',
        'contact_id_attribute' => '',
        'create_users' => '1',
    ],
```

**Comment by areis422** [comment]
Here's the error:

[Wed Sep 08 08:28:51.247700 2021] [proxy_fcgi:error] [pid 879497:tid 139643479652096] [client SOMEIP:15734] AH01071: Got error 'PHP message: #0 /var/www/eventum-3.10.6/src/Auth/Adapter/LdapAdapter.php(68): Eventum\\ServiceContainer::getLogger()\n#1 [internal function]: Eventum\\Auth\\Adapter\\LdapAdapter->__construct()\n#2 /var/www/eventum-3.10.6/src/Auth/Adapter/Factory.php(34): ReflectionClass->newInstanceArgs()\n#3 /var/www/eventum-3.10.6/src/Auth/Adapter/ChainAdapter.php(29): Eventum\\Auth\\Adapter\\Factory::create()\n#4 [internal function]: Eventum\\Auth\\Adapter\\ChainAdapter->__construct()\n#5 /var/www/eventum-3.10.6/src/Auth/Adapter/Factory.php(34): ReflectionClass->newInstanceArgs()\n#6 /var/www/eventum-3.10.6/lib/eventum/class.auth.php(473): Eventum\\Auth\\Adapter\\Factory::create()\n#7 /var/www/eventum-3.10.6/lib/eventum/class.auth.php(99): Auth::getAuthBackend()\n#8 /var/www/eventum-3.10.6/src/Controller/ListController.php(81): Auth::checkAuthentication()\n#9 /var/www/eventum-3.10.6/src/Controller/BaseController.php(80): Eventum\\Controller\\ListController->canAccess()\n#10 /var/www/eventum-3.10.6/htdocs/list.php(17): Eventum\\Controlle...PHP message: [2021-09-08 08:28:50] app.CRITICAL: Unable to instantiate auth adapter {"exception":"[object] (Error(code: 0): Call to a member function get() on null at /var/www/eventum-3.10.6/src/ServiceContainer.php:93)"} {"file":"/var/www/eventum-3.10.6/lib/eventum/class.auth.php","line":476,"class":"Auth","function":"getAuthBackend","memory_peak_usage":"4 MB","memory_usage":"4 MB","url":"/list.php","ip":"SOMEIP","http_method":"GET","server":"URL","referrer":"https://URL/login.php"}', referer: https://staging.eventum.genslerwi.com/login.php

**Comment by vladsf** [comment]
I guess [src/Auth/Adapter/LdapAdapter.php](https://github.com/eventum/eventum/blob/c23c0c1a786b6105753311f3f27ebf007c8b36b7/src/Auth/Adapter/LdapAdapter.php) is borken since 3.10.2  #898

**Comment by areis422** [comment]
Bump: Any update?

**Comment by vladsf** [comment]
@areis422 I tried to fix myself but failed so I changed to [`$this->logger = ServiceContainer::getLogger();` ](https://github.com/eventum/eventum/blob/c23c0c1a786b6105753311f3f27ebf007c8b36b7/src/Auth/Adapter/LdapAdapter.php#L68) and continued.

**Comment by shiyamk** [comment]
Thank you. it worked

**Comment by areis422** [comment]
@glensc This issue still exists in v3.10.9

**Comment by areis422** [comment]
Any update on the @glensc ?

**Comment by areis422** [comment]
Still currently exists in v3.10.12. Will this be released as part of v3.10.13?
